templates/data-provider/index.jsx

```diff
@@ -20,7 +20,8 @@ const DataProviderPageTemplate = React.forwardRef(
         <form ref={ref} onSubmit={handleSubmit}>
           <TextField
             id="data-provider-url"
-            type="url"
+            type="text"
+            pattern="^((ftp|http|https):\/\/)?([a-zA-Z0-9]+(\.[a-zA-Z0-9]+)+.*)$"
```

Comment:
I hope this regex I made is correct. (StackOverflow regexes haven't worked correctly for our use case)

I still think we should use type="url" though

Reply:
It's not correct. I will not cover the non-Latin domain name. Such domains names are encoded to Latin alphabet but if the user types it themselves, we will get non-Latin chars to check.

I am not sure about supporting `ftp://`, that's why I would suggest only `(https?:\/\/)?` in the beginning. I am afraid FTP is not supported by CHARS, there is no reason for it. @mcancellieri?

Reply:
I should probably drop ftp part

Reply:
the regex you are proposing makes "google" valid and that not true

Reply:
cannot we just stick with type="url" for the simplicity sake?

Reply:
But it is true, `localhost` is the first level domain. [about.google](https://about.google) is a valid domain too, however, `google` for some reason does not work.

It's not valid practically because it's too expensive to buy the first level domain for a regular university but not technically 🙂

---

I just could not say above ☝️ but I don't really understand what you mean

Reply:
Sorry I didn't notice you wanted to put that regex in the beginning. 

So what about this one:
```

```

Reply:
We should use type=url 😄

Reply:
I started looking into it. It's a pain. Some parts should be case-insensitive (the protocol), some are case-sensitive (the rest). Cyrillic domains should be encoded to `xn--` notation, special chars should be encoded using percentage notation.

I think, allowing to drop the protocol is not worth all this effort.

Reply:
Maybe we should run a test 
```js
if (!/^https?:\/\//i.test(value))
  value = `http://${value}`
```

before the validation.

However, if we cannot do so, we can drop this at all, sorry.